### ISSUE TITLE
[BUG] Use uniq before pluck

### ISSUE THREAD

**Issue by kvokka** [bbatsov/rubocop#3148]
In model i have
```
  def events_ids
    result = volunteer_roles.uniq.pluck(:event_id)
    result.any? ? result : nil
  end
```

`rubocop -a` returns

```
Offenses:

app/models/user.rb:157:47: C: [Corrected] Use uniq before pluck
    result = volunteer_roles.pluck(:event_id).uniq
                                              ^^^^
```

And in this case, it is wrong correction, because `uniq` or `distinct` methods return Array and `pluck` need Activerecord::Relation

**Comment by alexdowad** [comment]
@tjwp?

**Comment by kvokka** [comment]
I cannot replay it now, but as I see it is  closed :+1:

**Comment by IngoAlbers** [comment]
This is not fixed and should be reopened. 

See: https://github.com/bbatsov/rubocop/issues/3122#issuecomment-218126545

**Comment by jonas054** [comment]
I agree. It's not fixed.

**Comment by tjwp** [comment]
I don't think we can easily distinguish between `uniq` being called on `CollectionProxy` vs an `ActiveRecord::Relation` :(

I should either:

1. Make this cop disabled by default, remove autocorrection, and comment on when it may give a false positive. The cop has been useful for us, and may be helpful for others that do not typically use `uniq` or `distinct` on associations.
2. Remove the cop. I expect we'll continue to use it, but I'll just move it to a custom cop in our code.

Let me know what is preferred in situations like this.

**Comment by jonas054** [comment]
Maybe we can still "save" this cop by making it more conservative. I mean, we could start with some simple expressions that we believe will not generate many false positives. For example:

```ruby
MyModel.pluck(:col).uniq
```
where the receiver looks like a class. We let the cop only report such expressions.

But even so, removing (or disabling) auto-correct might be necessary.
